components/grid/drag-drop.md

```diff
@@ -0,0 +1,309 @@
+
+1. Use the `OnRowDrop` event to handle the drag and drop operations and modify the data source as per your business logic.
+
+
+## OnRowDrop Event
+
+The `OnRowDrop` event fires when the user drops a row into a new location. It allows you to manipulate your data collection based on where the user dropped the element. 
+
+### Event Arguments
+
+The `OnRowDrop` event provides an object of type `GridRowDropEventArgs<T>` to its event handler which exposes the following fields:
+
+* `Item` - an `object` that represents the dragged row. You can cast this object to your model class.
+
+* `DestinationItem` - an `object` that represents the row over which the `Item` is dropped. You can cast this object to your model class.
+
+* `DestinationItems` - `IEnumerable<T>` that represents a collection of all dragged items. 
+
+* `DropPosition` - an `enum` - its members allow you to determine the exact position of the dropped item relative to the position of the `DestinationItem`:
+    * `After`
+    * `Below`
+    * `Over`
+    
+* `DestinationGrid` - the reference of the Grid in which the row is dropped. This is applicable when you drag and drop rows between different grids. 
+
+## GridRowDraggableSettings
+
+The `GridRowDraggableSettings` is a child tag under the `<GridSettings>`, which is a child tag of the `<TelerikGrid>`. It exposes the following parameters:
+
+* `DragClueField` - `string` - defines which field will be used to render the drag clue text. By default, this parameter will take the value of the first bound column of the first dragged row. 
+
+You can find examples of its usage below.
+
+## Examples
+
+This section contains the following examples:
+
+* [Drag and Drop a Row in the same Grid](#drag-and-drop-a-row-in-the-same-grid)
+* [Drag and Drop a Row between Grids](#drag-and-drop-a-row-between-grids)
+* [Drag and Drop multiple Rows](#drag-and-drop-multiple-rows)
+
+### Drag and Drop a Row in the same Grid
+
+````Component
+@* Drag a row and drop it in the Grid. *@
+
+<TelerikGrid Data="@MyData" Height="400px"
+             Pageable="true" Sortable="true"
+             FilterMode="Telerik.Blazor.GridFilterMode.FilterRow"
+             Resizable="true" Reorderable="true"
+             RowDraggable="true"
+             OnRowDrop="@((GridRowDropEventArgs<SampleData> args) => OnRowDropHandler(args))">
+    <GridSettings>
+        <GridRowDraggableSettings DragClueField="@nameof(SampleData.Name)"></GridRowDraggableSettings>
+    </GridSettings>
+    <GridColumns>
+        <GridColumn Field="@(nameof(SampleData.Id))" Width="120px" />
+        <GridColumn Field="@(nameof(SampleData.Name))" Title="Employee Name" Groupable="false" />
+        <GridColumn Field="@(nameof(SampleData.Team))" Title="Team" />
+        <GridColumn Field="@(nameof(SampleData.HireDate))" Title="Hire Date" />
+    </GridColumns>
+</TelerikGrid>
+
+@code {
+    private void OnRowDropHandler(GridRowDropEventArgs<SampleData> args)
+    {
+        MyData.Remove(args.Item);
```

Comment:
I recommend you add a little bit of comments on what the code for data manipulations does and to denote that it is up to the app to implement this correctly as per its needs, even though it is mentioned in the text. Add it to every sample.

Also, consider disabling sorting and filtering because that will break the ordering logic which is rather naiive in this sample, consider linking the article about getting a row index to hint at people that the indexes are not so clear-cut. Also applies to all code snippets